packages/components/src/color-picker/alpha.js

```diff
@@ -34,34 +34,62 @@ import { noop } from 'lodash';
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Component, createRef } from '@wordpress/element';
+import { useState, useEffect, useRef } from '@wordpress/element';
 import { TAB } from '@wordpress/keycodes';
-import { pure } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
 import { calculateAlphaChange } from './utils';
 import KeyboardShortcuts from '../keyboard-shortcuts';
 
-export class Alpha extends Component {
-	constructor() {
-		super( ...arguments );
+export default function Alpha( { rgb, hsl, onChange = noop } ) {
+	const [ mouseDown, setMouseDown ] = useState( false );
+	const container = useRef();
 
-		this.container = createRef();
-		this.increase = this.increase.bind( this );
-		this.decrease = this.decrease.bind( this );
-		this.handleChange = this.handleChange.bind( this );
-		this.handleMouseDown = this.handleMouseDown.bind( this );
-		this.handleMouseUp = this.handleMouseUp.bind( this );
+	const rgbString = `${ rgb.r },${ rgb.g },${ rgb.b }`;
+	const gradient = {
+		background: `linear-gradient(to right, rgba(${ rgbString }, 0) 0%, rgba(${ rgbString }, 1) 100%)`,
+	};
+	const pointerLocation = { left: `${ rgb.a * 100 }%` };
+
+	function handleChange( e ) {
+		const change = calculateAlphaChange( e, { hsl }, container.current );
+		if ( change ) {
+			onChange( change, e );
+		}
+	}
+
+	function handleMouseUp() {
+		setMouseDown( false );
 	}
 
-	componentWillUnmount() {
-		this.unbindEventListeners();
+	function preventKeyEvents( event ) {
+		if ( event.keyCode === TAB ) {
+			return;
+		}
+		event.preventDefault();
 	}
 
-	increase( amount = 0.01 ) {
-		const { hsl, onChange = noop } = this.props;
+	useEffect( () => {
+		const { ownerDocument } = container.current;
+
+		function unbindEventListeners() {
+			ownerDocument.removeEventListener( 'mousemove', handleChange );
+			ownerDocument.removeEventListener( 'mouseup', handleMouseUp );
+		}
+
+		if ( mouseDown ) {
+			ownerDocument.addEventListener( 'mousemove', handleChange );
+			ownerDocument.addEventListener( 'mouseup', handleMouseUp );
+		} else {
+			unbindEventListeners();
```

Comment:
It doesn't seem right to be to remove listeners in the effect callback. That's what the clean up callback is for. In other words, if `mouseDown` changes, the clean up callback will run for you, so this code is not needed.

Reply:
See https://reactjs.org/docs/hooks-effect.html#effects-with-cleanup

Reply:
I removed the unnecessary code in daddou-ma/gutenberg@bf3acde